ingestion_server/ingestion_server/distributed_reindex_scheduler.py

```diff
@@ -87,6 +88,13 @@ def _prepare_workers():
         servers.append(server)
         ids.append(_id)
     log.info(f"Selected worker instances {servers}")
+
+    if worker_limit > 0:
+        log.info(f"Truncating worker instances under limit {worker_limit}")
+        servers = servers[:worker_limit]
+        ids = ids[:worker_limit]
+        log.info(f"Truncated worker instances {servers}")
```

Comment:
If we pass a worker limit _larger_ than the available worker count, what would be the side-effect here? Is there work that's already done that would get unnecessary re-done to recover from the error? Should we put an upper bound on `len(servers)`, or is it better to let it completely fail and make it obvious that there's a configuration error?

Reply:
Currently, slicing an array with a number larger than the length would have no effect at all and we would use all provisioned workers.

```python
servers = [1, 2, 3]
assert servers[:10] == servers
```

I wanted to avoid raising errors and failing the job so I felt it safer to let the job proceed with all available resources.

We could log something to the effect of f"Limit is higher than number of provisioned workers, using all {worker count} workers." if you feel that would be beneficial?

Reply:
> slicing an array with a number larger than the length would have no effect at all and we would use all provisioned workers

TIL! I assumed Python would through an index out of bounds error, but if there is no error from using a number larger than the array length, then my question is irrelevant. The approach you took makes perfect sense :slightly_smiling_face: